builtin/providers/dockercloud/resource_dockercloud_node_cluster.go

```diff
@@ -0,0 +1,255 @@
+package dockercloud
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/docker/go-dockercloud/dockercloud"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceDockercloudNodeCluster() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceDockercloudNodeClusterCreate,
+		Read:   resourceDockercloudNodeClusterRead,
+		Update: resourceDockercloudNodeClusterUpdate,
+		Delete: resourceDockercloudNodeClusterDelete,
+		Exists: resourceDockercloudNodeClusterExists,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"node_provider": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"size": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"region": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"disk": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"node_count": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+				ForceNew: false,
+			},
+			"state": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"tags": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: false,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
```

Comment:
Agreed, that should be easy enough to add, however, the upstream library doesn't expose that in the `NodeCreateRequest` struct. I can submit a PR to https://github.com/docker/go-dockercloud, but not sure how long it will take to get in.